rdr_service/genomic/genomic_job_controller.py

```diff
@@ -1815,18 +1814,37 @@ def _get_blocklist_str_attributes(all_values, map_values):
             logging.error(e)
             self.job_result = GenomicSubProcessResult.ERROR
 
-    @staticmethod
-    def update_member_file_record(manifest_type):
-        file_attr = None
-
-        attributes_map = {
-            GenomicManifestTypes.AW3_ARRAY: 'aw3ManifestFileId',
-            GenomicManifestTypes.AW3_WGS: 'aw3ManifestFileId'
-        }
-        if manifest_type in attributes_map.keys():
-            file_attr = attributes_map[manifest_type]
+    def process_manifest_record_updates(self, manifest_type, **kwargs):
+        if manifest_type not in [
+            GenomicManifestTypes.AW3_ARRAY, GenomicManifestTypes.AW3_WGS
+        ]:
+            return
 
-        return file_attr
+        if self.member_ids_for_update:
+            process_metrics = self.metrics_dao.get_bulk_metrics_for_process_update(
+                member_ids=self.member_ids_for_update,
+                pipeline_id=kwargs.get('pipeline_id')
+            )
+            members_to_update, metrics_to_update = [], []
+
+            for member_id in self.member_ids_for_update:
+                member_dict = {
+                    'id': member_id,
+                    'aw3ManifestFileId': kwargs.get('manifest_id')
+                }
+                members_to_update.append(member_dict)
+            self.member_dao.bulk_update(members_to_update)
+
+            for metric in process_metrics:
+                metric_dict = {
+                    'id': metric.id,
+                    'aw3ReadyFlag': 0,
+                    'aw3ManifestFileId': kwargs.get('manifest_id'),
```

Comment:
are you updating aw3ManifestJobRunID somewhere else?

Reply:
Good catch
79fbf81639d23dd58acc4dd74812464493f82346